### PR TITLE
add concat & concat_ws

### PR DESCRIPTION
**Pull request by nevermore3**

### CODE REVIEW

File: tests/tck/features/expression/FunctionCall.feature [hunk 1]
```diff
@@ -40,3 +40,73 @@ Feature: Function Call Expression
...
+      """
+      GO FROM "Tim Duncan" over like YIELD concat(like._src, $^.player.age, null, like.likeness) AS A
+      """
+    Then the result should be, in any order:
+      | A    |
+      | NULL |
+      | NULL |
+    When executing query:
+      """
+      match (a:player)-[b:serve]-(c:team{name: "Lakers"}) where a.age > 45   return  concat(a.name,c.name)
```
Comment (yixinglu) [inline tests/tck/features/expression/FunctionCall.feature:63]: please format the query prettily, such as:

```
MATCH ...
WHERE ...
RETURN ...
```

File: tests/tck/features/expression/FunctionCall.feature [hunk 2]
```diff
@@ -40,3 +40,73 @@ Feature: Function Call Expression
...
+      """
+    Then the result should be, in any order:
+      | concat_ws("@",a.name,NULL,"hello",b.likeness,c.name) |
+      | "Shaquile O'Neal@hello@Lakers"                       |
+    When executing query:
+      """
+      match (a:player)-[b:serve]-(c:team{name: "Lakers"}) where a.age > 45   return  concat_ws(1,a.name, NULL, "hello", b.likeness, c.name)
+      """
+    Then the result should be, in any order:
+      | concat_ws(1,a.name,NULL,"hello",b.likeness,c.name) |
```
Comment (yixinglu) [inline tests/tck/features/expression/FunctionCall.feature:104]: define an alias for the table header.

File: tests/tck/features/expression/FunctionCall.feature [hunk 3]
```diff
@@ -11,7 +11,7 @@ Feature: Function Call Expression
     When executing query:
       """
       YIELD sign(38) AS a, sign(-2) AS b, sign(0.421) AS c,
-            sign(-1.0) AS d, sign(0) AS e, sign(abs(-3)) AS f
+      sign(-1.0) AS d, sign(0) AS e, sign(abs(-3)) AS f
```
Comment (yixinglu) [inline tests/tck/features/expression/FunctionCall.feature:14]: prefer the former format

File: src/common/function/FunctionManager.cpp [hunk 4]
```diff
@@ -2145,6 +2154,91 @@ FunctionManager::FunctionManager() {
       return ds.rows[rowIndex][colIndex];
     };
   }
+  {
+    auto &attr = functions_["concat"];
+    attr.minArity_ = 1;
+    attr.maxArity_ = INT64_MAX;
+    attr.isPure_ = true;
+    attr.body_ = [](const auto &args) -> Value {
+      std::stringstream os;
```
[the remark this answers is not in the document]
Comment (nevermore3) [reply src/common/function/FunctionManager.cpp:2163]: if any parameter is null, return null in concat function
ignore parameter if it is null except first parameter, in concat_wc function

File: tests/tck/features/expression/FunctionCall.feature (hunk 3 above)
Comment (nevermore3) [reply tests/tck/features/expression/FunctionCall.feature]: done

File: tests/tck/features/expression/FunctionCall.feature (hunk 1 above)
Comment (nevermore3) [reply tests/tck/features/expression/FunctionCall.feature]: done

File: tests/tck/features/expression/FunctionCall.feature (hunk 2 above)
Comment (nevermore3) [reply tests/tck/features/expression/FunctionCall.feature]: done

File: src/common/function/FunctionManager.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (yixinglu) [reply src/common/function/FunctionManager.cpp:2163]: why different behaviors?

[the remark this answers is not in the document]
Comment (nevermore3) [reply src/common/function/FunctionManager.cpp:2163]: we refer to the implementation of concat & concat_ws in mysql

[the remark this answers is not in the document]
Comment (yixinglu) [reply src/common/function/FunctionManager.cpp:2163]: 👌

**Review by yixinglu — APPROVED** [review]

**Review by jievince — APPROVED** [review]
LGTM

**Pull request merged**